packages/hebao_v1/contracts/modules/core/ForwarderModule.sol

```diff
@@ -63,19 +65,21 @@ contract ForwarderModule is BaseModule
         bytes   memory signature
         )
         public
-        view
     {
-        require(
-            (to == from) ||
-            (to == controller.walletFactory()) ||
-            (to != address(this) && Wallet(from).hasModule(to)),
-            "INVALID_DESTINATION"
-        );
+        require(to != address(this), "INVALID_DESTINATION");
```

Comment:
Oh no wait, we need to disallow calls on **any** wallet address and stores here! This is still a module with special privileges on those contracts! Hmmm... I guess this is a drawback of having this as a module.

Reply:
oh.....

Reply:
Changing the check to `isModuleRegistered` instead of the wallet specific module check works I think. `callContract` can be used to call non-module contract functions (but not module functions), and meta txs can only be used to call module functions. Makes some sense. :)

Reply:
what if 

```
        require(
            to != address(this) &&
            controller.moduleRegistry().isModuleRegistered(to),
            "INVALID_DESTINATION"
        );
```

Reply:
@Brechtpd please review this part again. this is very important !!!!